### PR TITLE
feat: enrich permission selector entitlements with descriptions [WD-13431]

### PR DESCRIPTION
**Pull request by mas-who**
## Done

- Use entities metadata from `/1.0/metadata/configurations` to populate permission selector options
- This is built with backward compatibility in mind since `/1.0/metadata/configuration` endpoint does not exist for some older versions of LXD.

## QA

1. Run the LXD-UI:
    - On the demo server via the link posted by @webteam-app below. This is only available for PRs created by collaborators of the repo. Ask @mas-who or @edlerd for access.
    - With a local copy of this branch, [build and run as described in the docs](../CONTRIBUTING.md#setting-up-for-development).
2. Perform the following QA steps:
    - Go to Permissions -> Groups and create a group
    - For the new group, select "Manage Permissions" from the actions list to open up the permissions panel
    - View the options in the resource type selector, make sure all options are displayed correctly
    - Select a resource type, then select a resource. View the options in the entitlement selector, hover over an option and make sure that relevant description is displayed in the tooltip.

### CODE REVIEW

**Comment by piperdeck** [comment]
I'm not seeing tooltips in Safari 17.5 on macOS Sonoma 14.5 (Virtual machine)

https://github.com/user-attachments/assets/6ce22898-a21b-4242-9230-211a4e3fc469

Also not seeing them on Chromium 122.0.6261.128 or Firefox 128.0, both running on Ubuntu 22.04.4 LTS (Virtual Machine) 

https://github.com/user-attachments/assets/81463788-f2e9-4869-9c92-a3bd4d0d63c5

Note that other tooltips are showing

**Comment by piperdeck** [comment]
Also just tested Firefox and Chrome on macOS, same mac version as above


https://github.com/user-attachments/assets/80d3a3fd-1c0e-4d2a-a903-c0b9c795dfc3

**Comment by mas-who** [comment]
> Also just tested Firefox and Chrome on macOS, same mac version as above. Still not seeing the tooltips! Am I doing something wrong?
> 
>  CleanShot.2024-08-15.at.20.10.46.mp4

@piperdeck thanks for checking! I believe the issue is not due to browser version or variant, I think the permission entitlement descriptions are not showing due to the lxd version on our demo server being outdated. This feature is very new in lxd. Will update the lxd version on our demo server then it should work.

File: src/util/permissions.spec.ts [hunk 1]
```diff
@@ -110,16 +112,48 @@ describe("General util functions for permissions feature", () => {
...
+            description: "admin entitlement description",
+          },
+        ],
+      },
+    };
+
     const entitlementOptions = generateEntitlementOptions(
       resourceType,
       permissions,
+      metadata,
```
Comment (edlerd) [inline src/util/permissions.spec.ts:148]: Should we keep / add one test case for backwards compatibility, where metadata is null?

**Review by edlerd — COMMENTED** [review]
QA LGTM, some code observations below

File: src/types/config.d.ts [hunk 2]
```diff
@@ -23,6 +23,17 @@ export interface LxcConfigOptionCategories {
...
+  description: string;
+}
+
+export type LxdEntityMetadata = LxdEntityEntitlement[];
+
+export interface LxdEntities {
+  [entity: string]: LxdEntityMetadata;
+}
+
 export interface LxdConfigOptions {
```
[the remark this answers is not in the document]
Comment (mas-who) [reply src/types/config.d.ts:37]: Yeah good idea, will apply through out :+1:

File: src/types/config.d.ts [hunk 3]
```diff
@@ -23,6 +23,17 @@ export interface LxcConfigOptionCategories {
...
+export interface LxdEntityEntitlement {
+  name: string;
+  description: string;
+}
+
+export type LxdEntityMetadata = LxdEntityEntitlement[];
+
+export interface LxdEntities {
+  [entity: string]: LxdEntityMetadata;
+}
```
[the remark this answers is not in the document]
Comment (mas-who) [reply src/types/config.d.ts:35]: thanks this looks good!

File: src/util/permissions.spec.ts (hunk 1 above)
Comment (mas-who) [reply src/util/permissions.spec.ts:148]: done :+1:

File: src/util/permissions.tsx [hunk 4]
```diff
@@ -87,6 +94,30 @@ export const resourceTypeOptions = [
...
+  const options: typeof resourceTypeOptions = [
+    {
+      ...defaultOption,
+    },
+  ];
+
+  const resourceTypes = Object.keys(metadata.entities);
+  for (const resourceType of resourceTypes) {
+    const label = resourceType.split("_");
+    label[0] = label[0].charAt(0).toUpperCase() + label[0].slice(1);
```
[the remark this answers is not in the document]
Comment (mas-who) [reply src/util/permissions.tsx:111]: I found an existing helper `capitalizeFirstLetter` based on your comment haha, using that.

**Review by edlerd — APPROVED** [review]
LGTM :+1:
